classes/config_settings.php

```diff
@@ -0,0 +1,259 @@
+
+/**
+ * Data object containing SEB config settings with defaults if required.
+ *
+ * @package    quizaccess_seb
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace quizaccess_seb;
+
+defined('MOODLE_INTERNAL') || die();
+
+class config_settings {
+
+    /** @var string SETTINGS_TABLE DB table that contains quiz settings. */
+    const SETTINGS_TABLE = 'quizaccess_seb_quizsettings';
+
+    /** @var int $quizid Quiz ID of quiz the settings relate to. */
+    private $quizid;
+
+    /** @var \stdClass $settings Settings used to create class. */
+    private $settings;
+
+    /** @var int $existingsettingsid ID of existing settings in table. Defaults to -1 if none exist. */
+    private $existingsettingsid = -1;
+
+    /**
+     * config_settings constructor.
+     *
+     * @param int $quizid ID of quiz that the settings belong to.
+     * @param \stdClass|null $settings New quiz settings. May be provided by quiz settings form.
+     * @throws \dml_exception
+     * @throws \moodle_exception
+     */
+    public function __construct(int $quizid, \stdClass $settings = null) {
+        global $DB;
+
+        $this->quizid = $quizid;
+
+        // Try and retrieve existing settings, otherwise assign null.
+        $existingsettings = $DB->get_record(self::SETTINGS_TABLE, ['quizid' => $quizid], '*');
+        if (empty($existingsettings)) {
+            $existingsettings = new \stdClass();
+        } else {
+            $this->existingsettingsid = $existingsettings->id;
+        }
+
+        // If new settings are empty, set existing settings, otherwise merge existing settings with new settings.
+        if (empty((array) $settings)) {
+            $this->settings = $existingsettings;
+        } else {
+            $this->settings = $this->merge_new_and_existing_settings($this->strip_prefix($settings), $existingsettings);
+        }
+
+        // Associate settings with quiz.
+        $this->settings->quizid = $quizid;
+
+        // TODO: Process sebconfigtemplate into templateid.
+        $this->settings->templateid = 0;
+        unset($this->settings->sebconfigtemplate);
+
+        // Check to make sure all required settings are set up.
+        foreach ($this->get_required_settings_keys() as $name) {
+            if (!array_key_exists($name, $this->settings)) {
+                throw new \moodle_exception('missingrequiredsettings', 'quizaccess_seb', '', '', "Missing setting: $name");
+            }
+        }
+    }
+
+    /**
+     * Static constructor to create settings from existing quiz.
+     *
+     * @param string $quizid ID of quiz that the settings belong to.
+     * @return config_settings This instance.
+     *
+     * @throws \dml_exception
+     * @throws \moodle_exception
+     */
+    public static function with_quizid(string $quizid) {
```

Comment:
As per  https://docs.moodle.org/dev/Coding_style#Functions_and_Methods "Verbosity is encouraged: function names should be as illustrative as is practical to enhance understanding." Would suggest to pick a new name.